app/cli.py

```diff
@@ -653,3 +655,153 @@ def determine_dates_to_query_on_matomo(dates_in_database):
+      - github_daily_clones_count (a.k.a. DataLad download count estimation)
+
+    Warnings on major confounding factors in the number of dataset clones:
+      - the daily tests run on Circle CI create multiple clones per day (which
+      explains why there is always at least 1 unique clone counted per dataset
+      every day)
+      - the archiver and other automated script run by CONP contributes to the
+      number of clones every time a dataset is updated (including DATS and
+      README updates)
+      - CONP developers also contribute to the number of clones when testing,
+      updating or downloading datasets
+    """
+
+    from app import db
+    from app.models import GithubDailyClonesCount, GithubDailyViewsCount
+    from pathlib import Path
+    import git
+
+    datasetsdir = Path(app.config['DATA_PATH']) / 'conp-dataset'
+    try:
+        repo = git.Repo(datasetsdir)
+    except git.exc.InvalidGitRepositoryError:
+        repo = git.Repo.clone_from(
+            'https://github.com/CONP-PCNO/conp-dataset',
+            datasetsdir,
+            branch='master'
+        )
+
+    # loop through the list of submodules present in CONP-PCNO/conp-dataset.git
+    for submodule in repo.submodules:
+        sub_repo = submodule.url.replace('https://github.com/', '').replace('.git', '')
+        if sub_repo.startswith('CONP-PCNO/'):
+            # skip the repos under the CONP-PCNO organization as they are not datasets
+            continue
+
+        # query the GitHub analytics API for number of clones and views
+        daily_stat_dict = _get_repo_analytics(app, sub_repo)
+        if not daily_stat_dict:
+            continue
+
+        # get the list of dates already in the database for this repo
+        dates_in_db_dict = {}
+        db_clones_results = db.session.query(GithubDailyClonesCount.date).filter_by(repo=sub_repo).all()
+        dates_in_db_dict['clones'] = [row[0] for row in db_clones_results]
+        db_views_results = db.session.query(GithubDailyViewsCount.date).filter_by(repo=sub_repo).all()
+        dates_in_db_dict['views'] = [row[0] for row in db_views_results]
+
+        # loop through results returned by GitHub API calls and insert
+        # new data in the proper database table
+        for analytic_type in daily_stat_dict.keys():
+            if not daily_stat_dict[analytic_type]:
+                continue
+
+            for date in daily_stat_dict[analytic_type]:
+                if date in dates_in_db_dict[analytic_type]:
+                    # go to next date if there is already an entry for the date
+                    # for that repo in the database table
+                    continue
+
+                analytics_summary = None
+                if analytic_type == 'clones':
+                    analytics_summary = GithubDailyClonesCount()
+                elif analytic_type == 'views':
+                    analytics_summary = GithubDailyViewsCount()
+
+                if analytics_summary:
```

Comment:
Maybe log a warning that the `daily_stat_dict` was malformed in an `else` block after this.